tests/units/result_manager/conftest.py

```diff
@@ -56,30 +57,35 @@ def result_manager() -> ResultManager:
 
 
 @pytest.fixture
-def test_result_factory(device: AntaDevice) -> Callable[[int], TestResult]:
+def test_result_factory(device: AntaDevice) -> Callable[[int, int, bool, bool], TestResult]:
     """Return a anta.result_manager.models.TestResult object."""
     # pylint: disable=redefined-outer-name
 
-    def _create(index: int = 0) -> TestResult:
+    def _create(index: int = 0, nb_atomic_results: int = 0, distinct_tests: bool = False, distinct_devices: bool = False) -> TestResult:
         """Actual Factory."""
-        return TestResult(
-            name=device.name,
-            test=f"VerifyTest{index}",
+        test = FakeTestWithInput(device=device, inputs={"string": f"Test instance {index}"})
+        res = TestResult(
+            name=device.name if not distinct_devices else f"{device.name}{index}",
+            test=test.name if not distinct_tests else f"{test.name}{index}",
+            inputs=test.inputs,
             categories=["test"],
-            description=f"Verifies Test {index}",
+            description=test.description,
             custom_field=None,
         )
+        for i in range(nb_atomic_results):
+            res.add(description=f"{test.name}{index}AtomicTestResult{i}", inputs=test.inputs)
+        return res
 
     return _create
 
 
 @pytest.fixture
-def list_result_factory(test_result_factory: Callable[[int], TestResult]) -> Callable[[int], list[TestResult]]:
+def list_result_factory(test_result_factory: Callable[[int, int, bool, bool], TestResult]) -> Callable[[int, int, bool, bool], list[TestResult]]:
```

Comment:
to be honest if this is not needed anywhere else in the codebase I feel like the result_manager_factory could be enough without having to go through a list (but can be done in another PR)